holidays/countries/singapore.py

```diff
@@ -268,30 +277,20 @@ def _add_holiday(dt: date, hol: str) -> None:
         }
         if year in dates_fixed_obs:
             hol_date = date(year, *dates_fixed_obs[year])
-            self[hol_date] = "Deepavali"
+            _add_with_observed(hol_date, "Deepavali")
         else:
             hol_date = self.cnls.s_diwali_date(year)
-            self[hol_date] = "Deepavali* (*estimated)"
+            _add_with_observed(hol_date, "Deepavali* (*estimated)")
 
         # Christmas Day
-        self[date(year, DEC, 25)] = "Christmas Day"
+        _add_with_observed(
+            date(year, DEC, 25), "Christmas Day", +2 if year == 2039 else +1
+        )
 
         # Boxing day (up to and including 1968)
         if year <= 1968:
             self[date(year, DEC, 26)] = "Boxing Day"
 
-        # Check for holidays that fall on a Sunday and implement Section 4(2)
-        # of the Holidays Act: "if any day specified in the Schedule falls on
-        # a Sunday, the day next following not being itself a public holiday
-        # is declared a public holiday in Singapore."
-        if self.observed and year >= 1998:
-            for hol_date, hol_name in list(self.items()):
-                if hol_date.year == year and self._is_sunday(hol_date):
-                    in_lieu_date = hol_date + td(days=+1)
-                    while in_lieu_date.year == year and in_lieu_date in self:
-                        in_lieu_date += td(days=+1)
-                    _add_holiday(in_lieu_date, f"{hol_name} (Observed)")
-
         # special case (observed from previuos year)
         if self.observed and year == 2007:
             self[date(2007, JAN, 2)] = "Hari Raya Haji (Observed)"
```

Comment:
This one could also be declared in the special holidays section instead
```suggestion
        # special case (observed from previuos year)
        if self.observed and year == 2007:
            self[date(2007, JAN, 2)] = "Hari Raya Haji (Observed)"
```

Reply:
```python
        2007: ((JAN, 2, "Hari Raya Haji (Observed)"),),
```